Ceremony generate tool panics if syslog is not available

**Steps to reproduce**
Launch a container without syslog facilities and run the `generate` tool.
```
$ docker run -v /home/phil/work/boulder:/boulder -it letsencrypt/boulder-tools:go1.19.5_2023-02-02 bash

root@e7be8c2bb406:/# cd /boulder

root@e7be8c2bb406:/boulder# go run test/cert-ceremonies/generate.go 
panic: Unix syslog delivery error

goroutine 1 [running]:
github.com/letsencrypt/boulder/log.initialize()
        /boulder/log/log.go:113 +0x85
github.com/letsencrypt/boulder/log.Get.func1()
        /boulder/log/log.go:143 +0x25
sync.(*Once).doSlow(0x2b2?, 0xc00025fe00?)
        /usr/local/go/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:65
github.com/letsencrypt/boulder/log.Get()
        /boulder/log/log.go:141 +0x31
github.com/letsencrypt/boulder/cmd.Fail({0xc0003c0580, 0x2b2})
        /boulder/cmd/shell.go:282 +0x25
github.com/letsencrypt/boulder/cmd.FailOnError({0xa06820, 0xc0000244c0}, {0x958ea8?, 0x9?})
        /boulder/cmd/shell.go:296 +0xc5
main.main()
        /boulder/test/cert-ceremonies/generate.go:95 +0x278
exit status 2
```

If I configure a syslog server such as rsyslog, then it works as intended. Keep in mind that I'm in this container actively trying to debug a feature I'm working on which is how I found this panic.
```
root@ee99ad248071:/# apt update
root@ee99ad248071:/# apt install -y rsyslog
root@ee99ad248071:/# service rsyslog start
 * Starting enhanced syslogd rsyslogd 
root@ee99ad248071:/# service rsyslog status
 * rsyslogd is running
 * root@ee99ad248071:/boulder# go run test/cert-ceremonies/generate.go 
2023-02-10T23:29:23.273744+00:00Z ee99ad248071 unknown generate[1521]: 3 generate -fzC8QU ... ... ... ...
```

--- a/test/cert-ceremonies/generate.go
+++ b/test/cert-ceremonies/generate.go
@@ -73,8 +73,7 @@ func genCert(path string) error {
 }
 
 func main() {
-	logger := cmd.NewLogger(cmd.SyslogConfig{StdoutLevel: 6, SyslogLevel: -1})
-	logger.Info("Logging to stdout/stderr only, not syslog.")
+	_ = cmd.NewLogger(cmd.SyslogConfig{StdoutLevel: 6, SyslogLevel: -1})
 
 	// If one of the output files already exists, assume this ran once
 	// already for the container and don't re-run.
@@ -83,9 +82,9 @@ func main() {
 		fmt.Println("skipping certificate generation: already exists")
 		return
 	} else if err == nil && !loc.Mode().IsRegular() {
-		logger.Errf("statting %q: not a regular file", outputFile)
+		cmd.Fail(fmt.Sprintf("statting %q: not a regular file", outputFile))
 	} else if err != nil && !os.IsNotExist(err) {
-		logger.Errf("statting %q: %s", outputFile, err)
+		cmd.Fail(fmt.Sprintf("statting %q: %s", outputFile, err))
 	}
 	// Create SoftHSM slots for the root signing keys
 	rsaRootKeySlot, err := createSlot("root signing key (rsa)")

--- a/test/cert-ceremonies/generate.go
+++ b/test/cert-ceremonies/generate.go
@@ -10,6 +10,7 @@ import (
 	"text/template"
 
 	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
 )
 
 // createSlot initializes a SoftHSM slot and token. SoftHSM chooses the highest empty
@@ -73,7 +74,7 @@ func genCert(path string) error {
 }
 
 func main() {
-	_ = cmd.NewLogger(cmd.SyslogConfig{StdoutLevel: 6, SyslogLevel: -1})
+	blog.Set(blog.StdoutLogger(6))
 
 	// If one of the output files already exists, assume this ran once
 	// already for the container and don't re-run.

--- a/test/cert-ceremonies/generate.go
+++ b/test/cert-ceremonies/generate.go
@@ -1,9 +1,9 @@
+// generate.go is a helper utility for integration tests.
 package main
 
 import (
 	"errors"
 	"fmt"
-	"log"
 	"os"
 	"os/exec"
 	"regexp"
@@ -73,6 +73,9 @@ func genCert(path string) error {
 }
 
 func main() {
+	logger := cmd.NewLogger(cmd.SyslogConfig{StdoutLevel: 6, SyslogLevel: -1})
+	logger.Info("Logging to stdout/stderr only, not syslog.")
+
 	// If one of the output files already exists, assume this ran once
 	// already for the container and don't re-run.
 	outputFile := "/hierarchy/root-signing-pub-rsa.pem"
@@ -80,9 +83,9 @@ func main() {
 		fmt.Println("skipping certificate generation: already exists")
 		return
 	} else if err == nil && !loc.Mode().IsRegular() {
-		log.Fatalf("statting %q: not a regular file", outputFile)
+		logger.Errf("statting %q: not a regular file", outputFile)
 	} else if err != nil && !os.IsNotExist(err) {
-		log.Fatalf("statting %q: %s", outputFile, err)
+		logger.Errf("statting %q: %s", outputFile, err)
 	}
 	// Create SoftHSM slots for the root signing keys
 	rsaRootKeySlot, err := createSlot("root signing key (rsa)")

